app/code/Magento/InventoryCatalogSearch/Plugin/Model/Adapter/Mysql/Aggregation/DataProvider/SelectBuilderForAttribute/JoinStockConditionToSelect.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalogSearch\Plugin\Model\Adapter\Mysql\Aggregation\DataProvider\SelectBuilderForAttribute;
+
+use Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation\DataProvider\SelectBuilderForAttribute\JoinStockConditionToSelect
+    as LegacyStockConditionJoiner;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Select;
+use Magento\InventoryCatalog\Model\GetStockIdForCurrentWebsite;
```

Comment:
Need to remove dependency on private implementation of another module

Reply:
Fixed.